### PR TITLE
[AIR] Replace references of `to_tf` with `iter_tf_batches`

### PR DESCRIPTION
**Pull request by c21**
Signed-off-by: Cheng Su <scnju13@gmail.com>

## Why are these changes needed?

As title, we want to use `iter_tf_batches` in codebase other than `to_tf`.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by c21** [comment]
Didn't update the notebook yet - https://github.com/ray-project/ray/blob/master/doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb . Cannot run tensorflow on my M1 laptop now.

**Comment by c21** [comment]
It would be great if anyone tries to run some changed files here in their environment. I cannot run tensorflow yet, so these changes are NOT tested.

**Comment by richardliaw** [comment]
Can you also fix doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb ?

**Review by richardliaw — DISMISSED** [review]
1 comment

**Comment by c21** [comment]
@richardliaw - https://github.com/ray-project/ray/pull/27672#issuecomment-1208698214 I am trying to do it now.

**Review by clarkzinzow — APPROVED** [review]
LGTM once the comment is addressed.

>
> It would be great if anyone tries to run some changed files here in their environment. I cannot run tensorflow yet, so these changes are NOT tested.

These are all covered by our CI so if the CI passes it should all be good, right?

File: doc/source/data/doc_code/accessing_datasets.py
Comment (matthewdeng) [inline doc/source/data/doc_code/accessing_datasets.py:174]: This comment is no longer true.

Comment (matthewdeng) [reply doc/source/data/doc_code/accessing_datasets.py:174]: Also we may actually want to use concatenation for this example, so that it is closer to what consumers (e.g. Keras model) expect.

File: doc/source/ray-air/doc_code/tf_starter.py
Comment (matthewdeng) [inline doc/source/ray-air/doc_code/tf_starter.py:65]:
```suggestion
            for batch in dataset.iter_tf_batches(
                batch_size=batch_size, dtypes=tf.float32
            ):
                yield batch["x"], batch["y"]

        output_signature = (
            tf.TensorSpec(shape=(None, ), dtype=tf.float32),
            tf.TensorSpec(shape=(None), dtype=tf.float32),
        )
```

I'm a little rusty, but would this work since the model definition explicitly has this step?
```python
            # Add feature dimension, expanding (batch_size,) to (batch_size, 1).
            tf.keras.layers.Flatten(),
```

File: doc/source/data/doc_code/accessing_datasets.py
Comment (c21) [reply doc/source/data/doc_code/accessing_datasets.py:174]: Will remove this comment.

File: doc/source/ray-air/doc_code/tf_starter.py
Comment (c21) [reply doc/source/ray-air/doc_code/tf_starter.py:65]: I may have too little context to comment. anyone else has a better idea? @richardliaw

Comment (richardliaw) [reply doc/source/ray-air/doc_code/tf_starter.py:65]: I think in both cases it does the same thing. OK to just take Matt's suggestion as it seems to simplify the code abit.

**Review by richardliaw — COMMENTED** [review]
1 nit, otherwise looks good. The iter_tf_batches API change definitely isn't the best here.

**Comment by clarkzinzow** [comment]
@richardliaw Dismissing your requested changes as stale since the example has been updated!

**Comment by clarkzinzow** [comment]
Ah @richardliaw looks like we need an approving review from someone on the @ray-project/ray-docs team.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
